### ISSUE TITLE
🚀 Feature Request: Support Wildcard Domain Routings on Workers

### ISSUE THREAD

**Issue by hrueger** [cloudflare/workers-sdk#5568]
### Describe the solution

Hi, when currently using routes for Workers with wildcards, you get the following error:
```
✘ [ERROR] Cannot use "*.admin.dein-demo.shop" as a Custom Domain; wildcard operators (*) are not allowed
```
Example config:
```toml
routes = [
	{ pattern = "my.app", custom_domain = true },
    { pattern = "*.my.app", custom_domain = true }
]
```
When doing the same thing in the dashboard, it works fine, resulting in:
![grafik](https://github.com/cloudflare/workers-sdk/assets/18656830/15261eea-3a9a-4c33-bc20-0fc047570524)

However, when adding some routes in the dashboard manually but managing others in the toml file, the next deployment overwrites the manually added domains... :-(

**Comment by penalosa** [comment]
Custom Domains do not support wildcard routes, but in general routes in both `wrangler.toml` and the dashboard do support wildcard matching: https://developers.cloudflare.com/workers/configuration/routing/routes/

TLDR: Instead of setting `custom_domain = true`, set `zone_id` to the ID of your zone.

**Comment by CoreJa** [comment]
@hrueger Sorry to bother you, but I'm really stuck by adding a wildcard domain to my worker, even through the web GUI dashboard, any clues? It just says hostname invalid.

**Comment by hrueger** [comment]
Hi @CoreJa ,
Unfortunately, I've never managed to make that work and we've moved away from Cloudflare Workers 🤷‍♂️

**Comment by heretolya** [comment]
For anyone still looking for an answer: **wildcards do work with cloudflare workers**. 
You can make this work by configuring `wrangler.jsonc` + DNS settings.

1. Add a DNS AAAA record for your wildcard subdomain pointing to `100::` (e.g. *.example.com)
<img width="1128" height="62" alt="Image" src="https://github.com/user-attachments/assets/e409e7e1-e9a2-49bf-b871-888d7b09a21d" />

2. Configure your `wrangler.jsonc` (or `wrangler.toml`) to support a wildcard route as shown on the picture below
<img width="775" height="378" alt="Image" src="https://github.com/user-attachments/assets/ed0749f5-60ec-439b-ac94-140541d3edfd" />
